### PR TITLE
caching enabled status per logger

### PR DESCRIPTION
**Pull request by omg**
Can we have this to avoid rechecking rules on each call?

### CODE REVIEW

**Comment by Qix-** [comment]
Please explain what this does. What problem are you facing?

**Comment by omg** [comment]
Current implementation checks whether the namespace is enabled against the lists of regexes each time the debug instance is called. This change addresses this by caching the enabled flag in the debug instance to avoid reevaluating it each time. Since the library supports dynamically changing namespace configuration, the change also addresses this by caching reference to namespace config in the instance and comparing it against the current config (lazily). When the references are different the new config reference is replaces the previously cached one and the enabled flag reevaluated so nothing breaks but redundant regex checks are cut off.

File: src/common.js [hunk 1]
```diff
@@ -120,7 +121,17 @@ function setup(env) {
 		Object.defineProperty(debug, 'enabled', {
 			enumerable: true,
 			configurable: false,
-			get: () => enableOverride === null ? createDebug.enabled(namespace) : enableOverride,
+			get: () => {
+				if (enableOverride !== null) {
+					return enableOverride
```
Comment (Qix-) [inline src/common.js:126]:
```suggestion
					return enableOverride;
```

File: src/common.js [hunk 2]
```diff
@@ -120,7 +121,17 @@ function setup(env) {
...
+				}
+				if (namespacesCache !== createDebug.namespaces) {
+					namespacesCache = createDebug.namespaces;
+					enabledCache = createDebug.enabled(namespace);
+				}
+				return enabledCache;
+
```
Comment (Qix-) [inline src/common.js:133]:
```suggestion

				return enabledCache;
```

File: src/common.js [hunk 3]
```diff
@@ -60,6 +60,7 @@ function setup(env) {
 	function createDebug(namespace) {
 		let prevTime;
 		let enableOverride = null;
+		let namespacesCache, enabledCache;
```
Comment (Qix-) [inline src/common.js:63]:
```suggestion
		let namespacesCache;
		let enabledCache;
```

**Review by Qix- — CHANGES_REQUESTED** [review]
Sounds good, thanks!

Just a few small nitpicks and I'll merge.

**Comment by Qix-** [comment]
Released as `4.3.2` on the `beta` tag.

**Comment by diverdowngit** [comment]
juuhhyggbb

**Comment by diverdowngit** [comment]
2.6.9

**Pull request merged**